CTCScheduler.m

```diff
@@ -91,19 +94,29 @@ - (void)preventAppNap {
 }
 
 - (void)allowAppNap {
+    
   // Make sure we can keep running in the background if the system supports App Nap
-  if ([NSProcessInfo.processInfo respondsToSelector:@selector(beginActivityWithOptions:reason:)] && self.activityToken) {
+  // and that we have an activity token
+  if ([NSProcessInfo.processInfo respondsToSelector:@selector(beginActivityWithOptions:reason:)]
+      && self.activityToken) {
       [NSProcessInfo.processInfo endActivity:self.activityToken];
       self.activityToken = nil;
   }
 }
 
 - (void)updateAppNapStatus {
-    self.polling ? [self preventAppNap] : [self allowAppNap];
+    [self shouldPreventAppNap] ? [self preventAppNap] : [self allowAppNap];
```

Comment:
Here you're switching between preventing app nap + system sleep, and preventing neither. What we want is always prevent app nap, and only conditionally prevent system sleep.